frame/collective/src/lib.rs

```diff
@@ -218,7 +221,7 @@ decl_module! {
 			ensure!(Self::is_member(&who), Error::<T, I>::NotMember);
 
 			let proposal_hash = T::Hashing::hash_of(&proposal);
-			let ok = proposal.dispatch(RawOrigin::Member(who).into()).is_ok();
+			let ok = T::Dispatcher::dispatch(*proposal, RawOrigin::Member(who).into()).is_ok();
```

Comment:
I wonder if we can ever come up with an abstraction here to actually prevent one from calling `proposal.dispatch()` outside of the context of a `Dispatcher`. I think it should be possible by checking some global environmental value. But maybe it is not even needed.

Reply:
We could require the trait to accept a parameter of a type for which an instance can only be created from the `Dispatcher`'s module.

Reply:
The rewrite applies gavs suggestion in order to accomplish this.

Reply:
Do we have a `[should_panic]` test case around it? I'd love to see that 😄